src/App.tsx

```diff
@@ -1,25 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const total = 42;
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+
+  const items = getNumbers(1, total);
+  const totalPages = Math.ceil(items.length / perPage);
+
+  const indexOfLastItem = currentPage * perPage;
+  const indexOfFirstItem = indexOfLastItem - perPage;
+  const currentItems = items.slice(indexOfFirstItem, indexOfLastItem);
+
+  function handleSelectChange(event: React.ChangeEvent<HTMLSelectElement>) {
+    setPerPage(Number(event.target.value));
+    setCurrentPage(1);
+  }
+
+  function handleClick(page: number) {
+    setCurrentPage(page);
+  }
+
+  function showSubtitle(): string {
+    if (currentPage === totalPages) {
+      return `Page ${currentPage} (items ${indexOfFirstItem + 1} - ${items[items.length - 1]} of ${total})`;
```

Comment:
There's a potential off-by-one error in the display message when on the last page. The last item index should be 'items.length' instead of 'items[items.length - 1]', which gives the value of the last item rather than its index. Ensure the correct range is displayed.

Reply:
Ok